corehq/apps/api/resources/v0_5.py

```diff
@@ -655,13 +655,16 @@ class Meta(v0_4.GroupResource.Meta):
         always_return_data = True
 
     def serialize(self, request, data, format, options=None):
-        if not isinstance(data, dict):
+        if not isinstance(data, dict) and not self._is_list(request, data):
             if 'error_message' in data.data:
                 data = {'error_message': data.data['error_message']}
             elif request.method == 'POST':
                 data = {'id': data.obj._id}
         return self._meta.serializer.serialize(data, format, options)
 
+    def _is_list(self, request, data):
+        return request.method == 'PATCH' and isinstance(data, list)
```

Comment:
Why do we need to check the request method here?

Reply:
It's a conservative check. `serialize` branches on `not isinstance(data, dict)`, which is very broad. I didn't want this method to change behavior for request methods where it potentially should not apply.